src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java

```diff
@@ -274,6 +272,26 @@ public EntityBundle getEntityBundleForVersion(String entityId,
 			throw ExceptionUtil.convertSynapseException(e);
 		}
 	}
+	
+	@Override
+	public EntityBundlePlus getEntityBundlePlusForVersion(String entityId,
+			Long versionNumber, int partsMask) throws RestServiceException {
+		try {
+			org.sagebionetworks.client.SynapseClient synapseClient = createSynapseClient();
+			EntityBundle eb = synapseClient.getEntityBundle(entityId, partsMask);
```

Comment:
For Versionable entity types, it calls getEntityBundle() with the partsMask twice.  This is expensive.
I think I'd change it to do a synapseClient.getEntityById(entityId) up front.
if not Versionable then eb = synapseClient.getEntityBundle(entityId, partsMask),
otherwise eb = getEntityBundle(entityId, versionNumber, partsMask).